docs/tutorials/quick-start/quick-start-part-3.md

```diff
@@ -0,0 +1,808 @@
+  )
+}
+```
+
+### Sorting the Posts List
+
+Our `<PostsList>` is currently showing all the posts in the same order the posts are kept in the Redux store. Our example has the oldest post first, and any time we add a new post, it gets added to the end of the posts array. That means the newest post is always at the bottom of the page.
+
+Typically, social media feeds show the newest posts first, and you scroll down to see older posts. Even though the data is being kept oldest-first in the store, we can reorder the data in our `<PostsList>` component so that the newest post is first. In theory, since we know that the `state.posts` array is already sorted, we _could_ just reverse the list. But, it's better to go ahead and sort it ourselves just to be sure.
+
+Since `array.sort()` mutates the existing array, we need to make a copy of `state.posts` and sort that copy. We know that our `post.date` fields are being kept as date timestamp strings, and we can directly compare those to sort the posts in the right order:
+
+```jsx title="features/posts/PostsList.js"
+// Sort posts in reverse chronological order by datetime string
+//highlight-start
+const orderedPosts = posts.slice().sort((a, b) => b.date.localeCompare(a.date))
+
+const renderedPosts = orderedPosts.map((post) => {
+  //highlight-end
+  return (
+    <article className="post-excerpt" key={post.id}>
+      <h3>{post.title}</h3>
+      <div>
+        <PostAuthor userId={post.user} />
+        <TimeAgo timestamp={post.date} />
+      </div>
+      <p className="post-content">{post.content.substring(0, 100)}</p>
+      <Link to={`/posts/${post.id}`} className="button muted-button">
+        View Post
+      </Link>
+    </article>
+  )
+})
+```
+
+### Post Reaction Buttons
+
+We have one more new feature to add for this section. Right now, our posts are kind of boring. We need to make them more exciting, and what better way to do that than letting our friends add reaction emoji to our posts?
+
+We'll add a row of emoji reaction buttons at the bottom of each post in `<PostsList>` and `<SinglePostPage>`. Every time a user clicks one of the reaction buttons, we'll need to update a matching counter field for that post in the Redux store. Since the reaction counter data is in the Redux store, switching between different parts of the app should consistently show the same values in any component that uses that data.
+
+Like with post authors and timestamps, we want to use this everywhere we show posts, so we'll create a `<ReactionButtons>` component that takes a `post` as a prop. We'll start by just showing the buttons inside, with the current reaction counts for each button:
+
+```jsx title="features/posts/ReactionButtons.js"
+import React from 'react'
+
+const reactionEmoji = {
+  thumbsUp: '👍',
+  hooray: '🎉',
+  heart: '❤️',
+  rocket: '🚀',
+  eyes: '👀',
+}
+
+export const ReactionButtons = ({ post }) => {
+  const reactionButtons = Object.entries(reactionEmoji).map(([name, emoji]) => {
```

Comment:
It's a matter of style but usually, when I tech somebody React I'm trying to teach them the declarative way of thinking.

```ts
const reactionEmoji = {
  thumbsUp: "👍",
  hooray: "🎉",
  heart: "❤️",
  rocket: "🚀",
  eyes: "👀"
};
const ReactionButton = ({ emoji, children }) => {
  return (
    <button type="button" className="muted-button reaction-button">
      {reactionEmoji[emoji]} {children}
    </button>
  );
};
export const ReactionButtons = ({ post }) => {
  const {
    reactions: { hooray, heart, rocket, eyes, thumbsUp }
  } = post;
  
  return (
    <div>
      <ReactionButton emojiKey="thumbsUp">{thumbsUp}</ReactionButton>
      <ReactionButton emojiKey="hooray">{hooray}</ReactionButton>
      <ReactionButton emojiKey="heart">{heart}</ReactionButton>
      <ReactionButton emojiKey="rocket">{rocket}</ReactionButton>
      <ReactionButton emojiKey="eyes">{eyes}</ReactionButton>
    </div>
  );
};

// later
const onClick = (reaction) => {
  dispatch(reactionAdded({reaction, postId: post.id})
}
```

Just for my curiosity, you have chosen another way to write it, to decrease the number of lines?

Reply:
Yeah.  Either way works, I just didn't see a need to break up the component further.  It's perfectly fine to write larger components - not everything _has_ to be broken down to a tiny level.